core/src/main/scala/monocle/std/Option.scala

```diff
@@ -15,6 +15,11 @@ trait OptionFunctions {
   def none[A]: Prism[Option[A], Unit] =
     Prism[Option[A], Unit]{ case None => Some(()); case Some(_) => None }(_ => None)
 
+  def pOptionToDisjunction[A, B]: PIso[Option[A], Option[B], Unit \/ A, Unit \/ B] =
+    PIso[Option[A], Option[B], Unit \/ A, Unit \/ B](_.map(\/-(_)) getOrElse -\/(()))(_.toOption)
+
+  def optionToDisjunction[A]: Iso[Option[A], Unit \/ A] =
+    Iso[Option[A], Unit \/ A](_.map(\/-(_)) getOrElse -\/(()))(_.toOption)
```

Comment:
if you want, you can simply define `optionToDisjunction ` in terms of `pOptionToDisjunction `